internal/provider/credential_provider_resource.go

```diff
@@ -438,6 +438,10 @@ func (r *credentialProviderResource) Schema(_ context.Context, _ resource.Schema
 							}...),
 						},
 					},
+					"vault_private_network_access": schema.BoolAttribute{
+						Description: "Indicates if the Vault instance operates within a private network.",
+						Required:    true,
```

Comment:
Do we need to set a Default value as well?

Reply:
I thought about it but was not sure. I thought users would tend to forget setting it to true in case of a private network access if we have a default value.

If you think having a default value is more appropriate, I can change that.

Reply:
I'm not familiar with the vault use case, so I don't know what the best choice is here. @quantifex do you have a preference? Should we always force users to specify this value?